### ISSUE TITLE
Improve a11y of Modal component

### ISSUE THREAD

**Issue by julianguyen** [ifmeorg/ifme#2039]
# Description

From the [axe-core](https://github.com/dequelabs/axe-core), Deque's a11y linter:

```
Issue Description
Nested interactive controls are not announced by screen readers

Element location
#notificationsElement
Element source
<div id="notificationsElement" class="modalElement Modal__modalElement___2wDzo " role="button" tabindex="0">
To solve this issue, you need to...
Fix the following:
Element has focusable descendants
```

The element that lets us toggle the modal has `role="button"` but `tabIndex={0}`. We'll either want to get rid of this and back sure the children aren't focusable, or keep this and make sure the children are focusable.

---

Please assign yourself (via the Assignees dropdown), if you do want to work on this issue. Can't find yourself? You need to [join our organization](https://github.com/ifmeorg/ifme/wiki/Join-Our-Slack).

Check out our [Picking Up Issues](https://github.com/ifmeorg/ifme/wiki/Picking-Up-Issues) guide if you haven't already!
